### PR TITLE
Add Getting Started code samples

### PR DESCRIPTION
**Pull request by cmaneu**
- Add the .code-samples yaml files for supporting .NET Code samples in documentation
- Add all "Getting Started" code samples

See: #19

### CODE REVIEW

**Comment by cmaneu** [comment]
Hey Meilisearch team,
I've added a first series of code samples for the .NET client. I've used the new .NET 6 minimal API template, but I'm a bit worried it would not be natural for a lot of .NET developers. Happy to have your feedback on this.

**Comment by curquiza** [comment]
Hello @cmaneu 

I'm not an expert with .NET, so can you explain what is the difference between the new .NET 6 and what is commonly used?

Our goal is to reach the maximum of developers, so if it's not natural for the majority, I'm not sure this the right choice :)

And thank you so much for having started this PR :D

**Comment by cmaneu** [comment]
@curquiza we're aligned on the goal :)

## With .NET 6
.NET 6 is not yet released (will be before end of year). It comes with a new way of defining `Program.cs` with a more streamlined/minimal structure. 

```csharp
using System.Text.Json;
using Meilisearch;
 MeilisearchClient client = new MeilisearchClient("http://localhost:7700", "masterKey");
var options = new JsonSerializerOptions
{
    PropertyNameCaseInsensitive = true
};
string jsonString = await File.ReadAllTextAsync("movies.json");
var movies = JsonSerializer.Deserialize<List<Movie>>(jsonString, options);
var index = await client.GetOrCreateIndex("Movies");
await index.AddDocuments<Movie>(movies);
public class Movie
{
    public string Id { get; set; }
    public string Title { get; set; }
    public string Poster { get; set; }
    public string Overview { get; set; }
    public List<string> Genres { get; set; }
}
```

## Prior to .NET 6
The "classic" approach to writing programs involves adding namespaces, classes, etc... So the sample would like this.

```csharp
using System;
using System.Text.Json;
using Meilisearch;

namespace ConsoleApp1
{
    internal class Program
    {
        static void Main(string[] args)
        {
            MeilisearchClient client = new MeilisearchClient("http://localhost:7700", "masterKey");
            var options = new JsonSerializerOptions
            {
                PropertyNameCaseInsensitive = true
            };
            string jsonString = await File.ReadAllTextAsync("movies.json");
            var movies = JsonSerializer.Deserialize<List<Movie>>(jsonString, options);
            var index = await client.GetOrCreateIndex("Movies");
            await index.AddDocuments<Movie>(movies);
        }
    }

    public class Movie
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string Poster { get; set; }
        public string Overview { get; set; }
        public List<string> Genres { get; set; }
    }
}
```

I _assume_ that a .NET developer would be able to understand the .NET 6 sample, but will not be able to copy/paste it in their project. But a second pair of eyes on this assumption is more than welcome :)

**Comment by curquiza** [comment]
Thanks for this clear explanation 😁

I think this is better to not make our code samples with .NET 6! .NET 6 seems to bring great changes, but we would rather make the copy/paste work for most of the people rather than using the latest features of .NET.
We will consider changing our code samples using .NET 6 once this is widely used by the community.
What do you think? 🙂

**Comment by cmaneu** [comment]
Seems like a reasonable request :). I'm OOF tomorrow, but I'll update my PR - and remove the draft status - in the next days ;)

**Comment by curquiza** [comment]
@cmaneu, tell me when you are ready for a review :)

**Comment by cmaneu** [comment]
@curquiza It's should be ready for review. It's only the _getting started_ for now. I'll update some other pages with other PRs (and probably in few weeks)

**Comment by curquiza** [comment]
Ok, do you participate to hacktoberfest? I can wait for October to merge this one since it's a really valuable PR for MeiliSearch if you want :)

**Comment by cmaneu** [comment]
Hey @curquiza, Yes I'm registered to Hacktoberfest :) As we're (already) in October, feel free to merge it when you want ;)

File: .code-samples.meilisearch.yaml
Comment (curquiza) [inline .code-samples.meilisearch.yaml:111]:
```suggestion
            var index = await client.Index("Movies");
```

File: .code-samples.meilisearch.yaml
Comment (curquiza) [inline .code-samples.meilisearch.yaml:120]:
```suggestion
var index = await client.Index("Movies");
```

**Comment by curquiza** [comment]
I merge without the change applied, I will add them into my next PR :)

**Review by curquiza — APPROVED** [review]

**Comment by curquiza** [comment]
Thanks a lot @cmaneu, sorry for the delay

If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2021/), and you would like to receive a small gift from MeiliSearch too, please complete [this form](https://meilisearch.typeform.com/to/rPexR5gq).

**Comment by cmaneu** [comment]
Hello @curquiza, 
I don't see .NET / C# in the list of languages, but the docs has been rebuilt since this PR was merged.
Is that an expected behavior?
image

**Comment by curquiza** [comment]
Sorry, I thought I did a PR to the docs repo but I completely forgot
https://github.com/meilisearch/documentation/pull/1245

**Pull request merged**
